tests/Invoke-DbaDiagnosticQuery.Tests.ps1

```diff
@@ -1,16 +1,93 @@
-$CommandName = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
+# test ouput directory to confirm creation of test files
+$commandname = $MyInvocation.MyCommand.Name.Replace(".Tests.ps1", "")
 Write-Host -Object "Running $PSCommandpath" -ForegroundColor Cyan
-. "$PSScriptRoot\constants.ps1"
+. (Join-path (($PSScriptRoot, '.\' -ne '')[0]) "constants.ps1")  #if running in interactive console, cd to folder and you can run this without error
+write-host "loaded constants"
 
 Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
-    Context "Verifying output" {
+    BeforeAll {
+        $script:PesterOutputPath = "TestDrive:$commandName"
+
+
+        $database = "dbatoolsci_frk_$(Get-Random)"
+        $database2 = "dbatoolsci_frk_$(Get-Random)"
+        $server = Connect-DbaInstance -SqlInstance $script:instance2
+        $server.Query("CREATE DATABASE $database")
+        $server.Query("CREATE DATABASE $database2")
+
+    }
+    AfterAll {
+        $null = Get-DbaDatabase -SqlInstance $server -Databases @($database, $database2) | Remove-DbaDatabase -Confirm:$false
+        Remove-Item $script:PesterOutputPath -recurse -erroraction silentlycontinue  #clear test folder contents
+
+    }
+    AfterEach {
+        Remove-Item $script:PesterOutputPath -Recurse -erroraction silentlycontinue #clear test folder contents, doesn't need to fail if cleanup fails
+    }
+
+
+    Context "verifying output when running queries" {
         It "runs a specific query" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage' *>&1
-            $results.Name.Count | Should Be 1
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -QueryName 'Memory Clerk Usage'
+            @($results).Count | Should -Be 1
         }
         It "works with DatabaseSpecific" {
-            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific *>&1
-            $results.Name.Count -gt 10 | Should Be $true
+            $results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -DatabaseSpecific
+            @($results).count | Should -BeGreaterThan 10
         }
     }
-}
\ No newline at end of file
+
+    context "verifying output when exporting queries as files instead of running" {
+
+        It "exports queries to sql files without running" {
+            Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -ExportQueries -QueryName 'Memory Clerk Usage' -OutputPath $script:PesterOutputPath
+            @(Get-ChildItem -path $script:PesterOutputPath -filter *.sql).Count | Should -be 1
+        }
+
+        It "returns pscustomobject[] containing parsed queries" {
+            [System.Management.Automation.PSObject[]]$results = Invoke-DbaDiagnosticQuery -SqlInstance $script:instance2 -whatif
```

Comment:
why this type cohercition ? also, `-WhatIf`

Reply:
I did this check to ensure that objects were coming back with only pscustom object array. This was to ensure that no unbuffered output was creeping into the final result collection and impacting the returned array by having mixed types. For example, if someone put write-output type logic, this would break, so I thought I'd want to ensure the results came back without issue. Does this make sense?

Reply:
not really. got that you seem to like forcing types, but you're missing the big picture.
if someone puts write-output [pscustomobject] the function is screwed either way, isn't it ?
if you wanna check for creeping mistakes, check that every returned "thing" is an object with all the correct properties you expect

Reply:
Ok so you major difference in approach is that instead of worrying about the pipeline returning the exact same object type, you'd be more concerned with do all the properties I want exist? I might be just trying to be *too strongly typed* with a c# approach, and instead should design for properties existing, but not worry about object vs pscustomobject. Does that make sense?